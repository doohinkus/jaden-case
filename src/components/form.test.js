import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';

import Form from './form';

Enzyme.configure({ adapter: new Adapter() });

describe('Form', () => {
  it('should have a working change method', () => {
    const changeFunc = jest.fn();
    const submitFunc = jest.fn();
    const c = "change called";
    const output = shallow(
      <Form change={changeFunc} submit={submitFunc} />
    );
    output.find('input').simulate('change', c);
    expect(changeFunc).toBeCalledWith("change called");
  });

})
