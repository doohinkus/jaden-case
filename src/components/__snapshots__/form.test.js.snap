// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render the form 1`] = `
<form
  className="form-inline justify-content-center mt-3"
>
  <div
    className="form-group col-sm-8"
  >
    <input
      className="form-control w-100"
      id="inputText"
      placeholder="Tweet"
      type="text"
    />
  </div>
  <div
    className="form-group col-sm-4"
  >
    <Button
      text="Tweet"
    />
  </div>
</form>
`;
